mockOSApi.v2/Controllers/ProcessController.cs

```diff
@@ -12,20 +18,41 @@ public class ProcessController : Controller
     private readonly ILogger<ProcessController> _logger;
     private readonly IProcessService _processService;
 
-    public ProcessController(ILogger<ProcessController> logger, IProcessService processHandler)
+    private readonly IAuthentication _authorizationService;
+
+    public ProcessController(ILogger<ProcessController> logger, IProcessService processHandler, IAuthentication authorizationService)
     {
         _logger = logger;
         _processService = processHandler;
+        _authorizationService = authorizationService;
+    }
 
+    private string? FetchTokenFromHeaders()
+    {
+        string token;
+        try
+        {
+            token = HttpContext.Request.Headers["Authorization"].ToString().Split(" ")[1];
+        }
+        catch (System.IndexOutOfRangeException)
+        {
+            return null;
+        }
+        return token;
     }
 
     // GET /api/process
     [HttpGet]
     public string Index()
     {
+
         _logger.LogInformation("TBD [{0}]", DateTime.Now);
+        if (!_authorizationService.IsUserAuthorized(FetchTokenFromHeaders()))
```

Comment:
If you really want to, you can create yourself a custom Filter and Attribute.
Basically implementing the 'Authorize' attribute from scratch.
https://learn.microsoft.com/en-us/aspnet/core/mvc/controllers/filters?view=aspnetcore-8.0

Reply:
will work on this next